### PR TITLE
Add arbitrary rule data to ECP

### PR DESCRIPTION
**Pull request by zregvart**
Ref. https://issues.redhat.com/browse/HACBS-2042

### CODE REVIEW

File: api/config/appstudio.redhat.com_enterprisecontractpolicies.yaml [hunk 1]
```diff
@@ -83,6 +83,9 @@ spec:
                 rekorUrl:
                   description: URL of the Rekor instance. Empty string disables Rekor integration
                   type: string
+                ruleData:
+                  description: A json or yaml string containing rule data that will be visible to policy rules
```
Comment (simonbaird) [inline api/config/appstudio.redhat.com_enterprisecontractpolicies.yaml:87]: It's no longer a json or yaml string, right? Maybe:

```suggestion
                  description: Arbitrary rule data that will be visible to policy rules
```

**Review by simonbaird — APPROVED** [review]
Awesome.

File: api/config/appstudio.redhat.com_enterprisecontractpolicies.yaml [hunk 2]
```diff
@@ -83,6 +83,9 @@ spec:
                 rekorUrl:
                   description: URL of the Rekor instance. Empty string disables Rekor integration
                   type: string
+                ruleData:
+                  description: Arbitrary rule data that will be visible to policy rules
+                  x-kubernetes-preserve-unknown-fields: true
```
[the remark this answers is not in the document]
Comment (zregvart) [reply api/config/appstudio.redhat.com_enterprisecontractpolicies.yaml:88]: Can it every be an array?

[the remark this answers is not in the document]
Comment (lcarva) [reply api/config/appstudio.redhat.com_enterprisecontractpolicies.yaml:89]: I don't think it makes sense for it to be an array, a string, or anything other than an object.

**Comment by zregvart** [comment]
Let's merge this and try to work with it, I think it is safe to add now, and modify or remove if we need to in the future.

**Pull request merged**